mod/network.php

```diff
@@ -612,215 +716,154 @@ function network_content(App $a, $update = 0) {
 
-		switch ($order_mode) {
-			case 'received':
-				if ($last_received != '') {
-					$sql_extra3 .= sprintf(" AND $sql_table.`received` < '%s'", dbesc($last_received));
-					$a->set_pager_page(1);
-					$pager_sql = sprintf(" LIMIT %d, %d ",intval($a->pager['start']), intval($a->pager['itemspage']));
-				}
-				break;
-			case 'commented':
-				if ($last_commented != '') {
-					$sql_extra3 .= sprintf(" AND $sql_table.`commented` < '%s'", dbesc($last_commented));
-					$a->set_pager_page(1);
-					$pager_sql = sprintf(" LIMIT %d, %d ",intval($a->pager['start']), intval($a->pager['itemspage']));
-				}
-				break;
-			case 'created':
-				if ($last_created != '') {
-					$sql_extra3 .= sprintf(" AND $sql_table.`created` < '%s'", dbesc($last_created));
-					$a->set_pager_page(1);
-					$pager_sql = sprintf(" LIMIT %d, %d ",intval($a->pager['start']), intval($a->pager['itemspage']));
-				}
-				break;
-			case 'id':
-				if (($last_id > 0) && ($sql_table == "`thread`")) {
-					$sql_extra3 .= sprintf(" AND $sql_table.`iid` < '%s'", dbesc($last_id));
-					$a->set_pager_page(1);
-					$pager_sql = sprintf(" LIMIT %d, %d ",intval($a->pager['start']), intval($a->pager['itemspage']));
-				}
-				break;
+	$parents_arr = array();
+	$parents_str = '';
+	$date_offset = "";
+
+	if (dbm::is_result($r)) {
+		foreach ($r as $rr) {
+			if (!in_array($rr['item_id'],$parents_arr)) {
+				$parents_arr[] = $rr['item_id'];
+			}
 		}
 
-		// Fetch a page full of parent items for this page
-		if ($update) {
-			if (get_config("system", "like_no_comment"))
-				$sql_extra4 = " AND `item`.`verb` = '".ACTIVITY_POST."'";
-			else
-				$sql_extra4 = "";
-
-			$r = q("SELECT `item`.`parent` AS `item_id`, `item`.`network` AS `item_network`, `contact`.`uid` AS `contact_uid`
-				FROM $sql_table $sql_post_table INNER JOIN `contact` ON `contact`.`id` = `item`.`contact-id`
-				AND (NOT `contact`.`blocked` OR `contact`.`pending`)
-				WHERE `item`.`uid` = %d AND `item`.`visible` AND NOT `item`.`deleted` $sql_extra4
-				AND NOT `item`.`moderated` AND `item`.`unseen`
-				$sql_extra3 $sql_extra $sql_nets
-				ORDER BY `item_id` DESC LIMIT 100",
-				intval(local_user())
-			);
-		} else {
-			$r = q("SELECT `thread`.`iid` AS `item_id`, `thread`.`network` AS `item_network`, `contact`.`uid` AS `contact_uid`
-				FROM $sql_table $sql_post_table STRAIGHT_JOIN `contact` ON `contact`.`id` = `thread`.`contact-id`
-				AND (NOT `contact`.`blocked` OR `contact`.`pending`)
-				WHERE `thread`.`uid` = %d AND `thread`.`visible` AND NOT `thread`.`deleted`
-				AND NOT `thread`.`moderated`
-				$sql_extra2 $sql_extra3 $sql_extra $sql_nets
-				ORDER BY $sql_order DESC $pager_sql ",
-				intval(local_user())
-			);
+		$parents_str = implode(", ", $parents_arr);
+
+		// splitted into separate queries to avoid the problem with very long threads
+		// so always the last X comments are loaded
+		// This problem can occur expecially with imported facebook posts
```

Comment:
Is it then still relevant?

Reply:
Not with Facebook, but of course there could be long Diaspora- or Friendica-Threads as well.